### PR TITLE
Generate server diff in CI

### PR DESCRIPTION
**Pull request by crisidev**
## Motivation and Context
Generate server diff in CI

## Description
This should generate diffs from the codegen-server-test folder

## Testing
Let's see what the CI will say 😛 

## Checklist
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated `aws/SDK_CHANGELOG.md` if applicable

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tools/mk-generated.sh
Comment (rcoh) [inline tools/mk-generated.sh:30]: something is messy in this file, I think it ended up mixed tabs and spaces somehow, do you mind just changing the file to spaces

**Review by rcoh — APPROVED** [review]
LGTM! lets see if it works 😅

File: tools/mk-generated.sh
Comment (crisidev) [reply tools/mk-generated.sh:30]: Yep, I'll retab it.

**Comment by crisidev** [comment]
> LGTM! lets see if it works sweat_smile

Eh, I had to change the generation to avoid running tests and cargo doc :)

**Comment by crisidev** [comment]
> A new generated diff is ready to view: [__generated-main...__generated-server_diff](https://github.com/awslabs/smithy-rs/compare/__generated-main...__generated-server_diff)

Still a folder to remove 😞

**Comment by crisidev** [comment]
Ok, it works. I'll wait for ci and merge

**Pull request merged**
